pkg/build/graph/edges.go

```diff
@@ -55,24 +57,37 @@ func AddInputOutputEdges(g osgraph.MutableUniqueGraph, node *buildgraph.BuildCon
 		g.AddEdge(in, node, BuildInputEdgeKind)
 	}
 
+	sourceImages := []*kapi.ObjectReference{}
 	from := buildutil.GetImageStreamForStrategy(node.BuildConfig.Spec.Strategy)
 	if from != nil {
-		switch from.Kind {
+		sourceImages = append(sourceImages, from)
```

Comment:
My understanding is that even if there is no ICT, your from establishes a dependency on the image and it should be included in the chain. At least that's the way it's working now.